common/primitives/src/parquet.rs

```diff
@@ -0,0 +1,42 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use sp_std::prelude::*;
+
+/// Importing all base types
+pub mod base;
+/// Importing all numeric types
+pub mod numeric;
+/// Importing all string types
+pub mod string;
+/// Importing all temporal types
+pub mod temporal;
+/// Importing all labels
+pub mod types;
+/// Importing all compression codec types
+pub mod compression_codec;
+
+use crate::parquet::types::ParquetType;
+use crate::parquet::compression_codec::CompressionCodec;
+
+/// The model for Parquet data
+#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq, MaxEncodedLen, Serialize, Deserialize)]
+pub struct ParquetModel {
+  /// Parquet type labels
+  _type: ParquetType,
+  /// Compression for bloom filter
+  compression: CompressionCodec,
+  /// Whether or not to use a bloom filter
+  bloom_filter: bool,
+}
+
+impl ParquetModel {
+  /// Creates instance of struct
+  pub fn new(_type: ParquetType, compression: CompressionCodec, bloom_filter: bool) -> ParquetModel {
+    ParquetModel {
```

Comment:
You do not need to repeat parameter name and struct field, since they are the same
https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-the-field-init-shorthand